src/_igraph/graphobject.c

```diff
@@ -4118,6 +4118,85 @@ PyObject *igraphmodule_Graph_closeness(igraphmodule_GraphObject * self,
   return list;
 }
 
+/** \ingroup python_interface_graph
+ * \brief Calculates the harmonic centrality of some vertices in a graph.
+ * \return the harmonic centralities as a list (or a single float)
+ * \sa igraph_closeness
+ */
+PyObject *igraphmodule_Graph_harmonic_centrality(igraphmodule_GraphObject * self,
+                                       PyObject * args, PyObject * kwds)
+{
+  static char *kwlist[] = { "vertices", "mode", "cutoff", "weights",
+                "normalized", NULL };
+  PyObject *vobj = Py_None, *list = NULL, *cutoff = Py_None,
+           *mode_o = Py_None, *weights_o = Py_None, *normalized_o = Py_True;
+  igraph_vector_t res, *weights = 0;
+  igraph_neimode_t mode = IGRAPH_ALL;
+  int return_single = 0;
+  igraph_vs_t vs;
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "|OOOOO", kwlist, &vobj,
+      &mode_o, &cutoff, &weights_o, &normalized_o))
+    return NULL;
+
+  if (igraphmodule_PyObject_to_neimode_t(mode_o, &mode)) return NULL;
+  if (igraphmodule_PyObject_to_vs_t(vobj, &vs, &self->g, &return_single, 0)) {
+    igraphmodule_handle_igraph_error();
+    return NULL;
+  }
+
+  if (igraph_vector_init(&res, 0)) {
+    igraph_vs_destroy(&vs);
+    return igraphmodule_handle_igraph_error();
+  }
+
+  if (igraphmodule_attrib_to_vector_t(weights_o, self, &weights,
+      ATTRIBUTE_TYPE_EDGE)) {
+    igraph_vs_destroy(&vs);
+    igraph_vector_destroy(&res);
+    return NULL;
+  }
+
+  if (cutoff == Py_None) {
+    if (igraph_harmonic_centrality(&self->g, &res, 0, 0, vs, mode, weights,
+             PyObject_IsTrue(normalized_o))) {
+      igraph_vs_destroy(&vs);
+      igraph_vector_destroy(&res);
+      if (weights) { igraph_vector_destroy(weights); free(weights); }
+      igraphmodule_handle_igraph_error();
+      return NULL;
+    }
+  } else if (PyNumber_Check(cutoff)) {
+    PyObject *cutoff_num = PyNumber_Float(cutoff);
+    if (cutoff_num == NULL) {
+      igraph_vs_destroy(&vs); igraph_vector_destroy(&res);
+      return NULL;
```

Comment:
`weights` should be destroyed here. It might also be a bug in the closeness centrality if this part of the code was copied from there.

Reply:
Just checked; the problem also appears there (but not for betweenness or edge betweenness).